src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,789 @@
+            } else {
+                // update the file name mapping
+                for (String inputFileName : inputFileNames) {
+
+                    // filenames returned should never be blank/empty/null
+                    if (StringUtils.isBlank(inputFileName)) {
+                        criticalError("One of the file names returned as ready to process [" + inputFileName
+                                + "] was blank.  This should not happen, so throwing an error to investigate.");
+                    }
+
+                    inputFileTypeMap.put(inputFileName, batchInputFileType);
+                }
+            }
+        }
+
+        return inputFileTypeMap;
+    }
+
+    /**
+     * Processes a single ACH input file.
+     */
+    // Portions of this method are based on the code and logic from CustomerLoadServiceImpl.loadFile
+    protected List<String> loadACHBatchDetailFile(String inputFileName, BatchInputFileType batchInputFileType) {
+        List<String>failedRowsErrors = new ArrayList<String>();
+        
+        byte[] fileByteContent = safelyLoadFileBytes(inputFileName);
+        
+        LOG.info("loadACHBatchDetailFile: Attempting to parse the file.");
+        
+        Object parsedObject = null;
+        try {
+            parsedObject = batchInputFileService.parse(batchInputFileType, fileByteContent);
+        } catch (ParseException e) {
+            String errorMessage = "loadACHBatchDetailFile: Error parsing batch file: " + e.getMessage();
+            LOG.error(errorMessage, e);
+            throw new RuntimeException(errorMessage);
+        }
+        
+        if (!(parsedObject instanceof List)) {
+            String errorMessage = "loadACHBatchDetailFile: Parsed file was not of the expected type.  Expected [" + List.class + "] but got [" + parsedObject.getClass() + "].";
+            criticalError(errorMessage);
+        }
+
+        List<PayeeACHAccountExtractDetail> achDetails = (List<PayeeACHAccountExtractDetail>) parsedObject;
+        
+        for (PayeeACHAccountExtractDetail achDetail : achDetails) {
+            String error = processACHBatchDetail(achDetail);
+            if (ObjectUtils.isNotNull(error)) {
+                failedRowsErrors.add(error);
+            }
+        }
+        
+        return failedRowsErrors;
+    }
+
+    /**
+     * Processes a single ACH batch detail, and routes a Payee ACH Account maintenance document accordingly.
+     */
+    protected String processACHBatchDetail(PayeeACHAccountExtractDetail achDetail) {
+        LOG.info("processACHBatchDetail: Starting processACHBatchDetail for: " +achDetail.getLogData());
+        String processingError = null;
+        Person payee = personService.getPersonByPrincipalName(achDetail.getNetID());
+
+        processingError = validateACHBatchDetail(achDetail, payee);
+        if (ObjectUtils.isNotNull(processingError) && StringUtils.isNotBlank(processingError)) {

```

Comment:
isNotBlank will do the null check here, so you don't need the ObjectUtils.isNotNull in addition